File: facade/src/main/java/org/tdf/sunflower/types/VMResult.java
package org.tdf.sunflower.types;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.tdf.common.types.Uint256;
import org.tdf.common.util.ByteUtil;
import org.tdf.common.util.HexBytes;
import org.tdf.sunflower.state.Address;

import java.util.Collections;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class VMResult {
    public static final
    VMResult EMPTY = new VMResult(0, Address.empty(), ByteUtil.EMPTY_BYTE_ARRAY, Collections.emptyList(), Uint256.ZERO);

    long gasUsed;
    HexBytes contractAddress;
    byte[] executionResult;
    List<LogInfo> logs;
    Uint256 fee;
}

File: facade/src/main/java/org/tdf/sunflower/types/VMResult.kt
package org.tdf.sunflower.types

import org.tdf.common.types.Uint256
import org.tdf.common.util.HexBytes

class VMResult(
    val gasUsed: Long = 0,
    val executionResult: HexBytes = HexBytes.empty(),
    val logs: List<LogInfo> = emptyList(),
    val fee: Uint256 = Uint256.ZERO
) {

    companion object {
        val EMPTY = VMResult()
    }
}